src/agri_ai/core/agent_pool.py

```
@@ -0,0 +1,320 @@
+            
+            self.total_agents_created += 1
+            
+            logger.info(f"Created new agent for user {user_id} (total: {len(self.agents)})")
+            return agent
+            
+        except Exception as e:
+            logger.error(f"Failed to create agent for user {user_id}: {e}")
+            raise AgentProcessingError(
+                f"エージェントの作成に失敗しました: {str(e)}",
+                context={"user_id": user_id}
+            )
```

Ruff (0.11.9)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)